### PR TITLE
Add matmul_precision option for dq.set_precision

### PR DESCRIPTION
**Pull request by gautierronan**

### CODE REVIEW

File: dynamiqs/utils/jax_utils.py [hunk 1]
```diff
@@ -132,3 +140,15 @@ def set_precision(precision: Literal['simple', 'double']):
             f"Argument `x` should be a string 'simple' or 'double', but is"
             f" '{precision}'."
         )
+
+    if matmul_precision == 'low':
+        jax.config.update('jax_default_matmul_precision', 'bfloat16')
+    elif matmul_precision == 'high':
+        jax.config.update('jax_default_matmul_precision', 'bfloat16_3x')
+    elif matmul_precision == 'highest':
+        jax.config.update('jax_default_matmul_precision', 'float32')
```
Comment (abocquet) [inline dynamiqs/utils/jax_utils.py:149]: It seems strange that we cannot go to float64 for matrix product no? What is the point of f64 then?

Comment (gautierronan) [reply dynamiqs/utils/jax_utils.py:149]: We can, it's just that this API is very poorly written from Jax. Float32 actually means float32 or float64 depending on the global precision. Figure out! 🤷

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Great addition!!

I think it'd be better to have two different functions `set_precision` and `matmul_precision`. It will make it easier to change these parameters independently (e.g. in a loop). `matmul_precision` is also more niche than `set_precision` which would be used pretty often, it's good to keep it simple.

Another option would be to have a `dq.set(device=?, precision=?, matmul_precision=?)` with only optional arguments, but the doc would get a bit crazy, so I'd prefer the current version (with one function per option).

File: dynamiqs/utils/jax_utils.py (hunk 1 above)
Comment (abocquet) [reply dynamiqs/utils/jax_utils.py:149]: We should comment that somewhere

**Review by pierreguilmin — APPROVED** [review]
Very nice!! I've added commits with a few suggestions.

File: dynamiqs/utils/jax_utils.py (hunk 1 above)
Comment (pierreguilmin) [reply dynamiqs/utils/jax_utils.py]: See https://jax.readthedocs.io/en/latest/jax.lax.html#jax.lax.Precision, I've switched the key to `'highest'` which is not ambiguous.

**Pull request merged**